### ISSUE TITLE
Documentation polishing

### ISSUE THREAD

**Issue by deliciouslytyped** [chpatrick/clang-lens#21]
It would be a nice touch if https://clang.llvm.org/docs/index.html#using-clang-as-a-library was mentioned somewhere for people that don't already know how to use libclang.

**Comment by chpatrick** [comment]
That page is talking about the Clang C++ library though, which is different.

**Comment by deliciouslytyped** [comment]
Right, but that's what gives the underlying semantics of this binding? or did I miss some documents explaining in detail how this library is supposed to work without knowing how the C++ lib works?
